.github/workflows/lint-and-tests.yml

```diff
@@ -0,0 +1,22 @@
+name: Run lint and tests on PR
+on:
+  pull_request:
+      types: [opened, edited]
+      branches:
+      - main
```

Comment:
will the PR created to other branches trigger this hook?

Reply:
No, I made it only for main branch. Should it runs for other branches?

Reply:
I suppose yes, 'cause ensuring the quality of the code, regardless of which branch to merge it into, is a rule of thumb

Reply:
Done